app/preference.go

```diff
@@ -49,8 +47,14 @@ func (a *App) UpdatePreferences(userId string, preferences model.Preferences) *m
 	}
 
 	if err := a.Srv().Store.Preference().Save(&preferences); err != nil {
-		err.StatusCode = http.StatusBadRequest
-		return err
+		var appErr *model.AppError
+		switch {
+		case errors.As(err, &appErr):
+			appErr.StatusCode = http.StatusBadRequest
+			return appErr
```

Comment:
nit: There's only one case where it returns an `AppError` and is when is calling `IsValid` and the `IsValid` for `Preference` always return a `http.StatusBadRequest` so I think the overwrite of the status here is not needed.

Reply:
Will remove that overwrite then, thanks!